packages/belos/tpetra/src/BelosMultiVecTraits_Tpetra.hpp

```diff
@@ -409,6 +409,12 @@ namespace Belos {
         mv.multiply (Teuchos::NO_TRANS, Teuchos::NO_TRANS,
                      alpha, A, B_mv, beta);
       }
+      Kokkos::fence();  // Belos with Thyra's MvTimesMatAddMv allowed failures
+                        // when fence was not applied after mv.multiply; 
+                        // adding the fence fixed the tests in Thyra.  
+                        // Out of an abundance of caution (and with blessing 
+                        // from @hkthorn), we add the fence here as well.  
+                        // #8821 KDD
```

Comment:
Even if this passes the PR test, I do not understand why the fence is required here. The Tpetra kernels is launched and if getLocalView interface is properly used, it will do lazy sync when it is necessary whe the getLocalView is used next.